controllers/apps/transformer_rbac.go

```diff
@@ -0,0 +1,148 @@
+	"github.com/apecloud/kubeblocks/internal/controller/graph"
+	ictrltypes "github.com/apecloud/kubeblocks/internal/controller/types"
+)
+
+// RBACTransformer puts the rbac at the beginning of the DAG
+type RBACTransformer struct{}
+
+const PGTYPE = "postgresql"
+
+var _ graph.Transformer = &RBACTransformer{}
+
+func (c *RBACTransformer) Transform(ctx graph.TransformContext, dag *graph.DAG) error {
+	transCtx, _ := ctx.(*ClusterTransformContext)
+	cluster := transCtx.Cluster
+	if cluster.IsDeleting() {
+		return nil
+	}
+
+	root, err := ictrltypes.FindRootVertex(dag)
+	if err != nil {
+		return err
+	}
+
+	serviceAccount, err := builder.BuildServiceAccount(cluster)
+	if err != nil {
+		return err
+	}
+	saVertex := ictrltypes.LifecycleObjectCreate(dag, serviceAccount, root)
+
+	role, err := builder.BuildRole(cluster)
+	if err != nil {
+		return err
+	}
+
+	completeRoleRules(transCtx, role)
+	roleVertex := ictrltypes.LifecycleObjectCreate(dag, role, root)
+
+	roleBinding, err := builder.BuildRoleBinding(cluster)
+	if err != nil {
+		return err
+	}
+	rbVertex := ictrltypes.LifecycleObjectCreate(dag, roleBinding, root)
+	dag.Connect(rbVertex, roleVertex)
+	dag.Connect(rbVertex, saVertex)
+
+	statefulSetVertices := ictrltypes.FindAll[*appsv1.StatefulSet](dag)
+	for _, statefulSetVertex := range statefulSetVertices {
+		// rbac must be created before statefulset
+		dag.Connect(statefulSetVertex, rbVertex)
+	}
+
+	deploymentVertices := ictrltypes.FindAll[*appsv1.Deployment](dag)
+	for _, deploymentVertex := range deploymentVertices {
+		// rbac must be created before deployment
+		dag.Connect(deploymentVertex, rbVertex)
+	}
+	return nil
+}
+
+func completeRoleRules(transCtx *ClusterTransformContext, role *rbacv1.Role) {
+	rules := []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{""},
+			Resources: []string{"events"},
+			Verbs:     []string{"create"},
+		},
+		{
+			APIGroups: []string{"dataprotection.kubeblocks.io"},
+			Resources: []string{"backups/status"},
+			Verbs:     []string{"get", "update", "patch"},
+		},
+		{
+			APIGroups: []string{"dataprotection.kubeblocks.io"},
+			Resources: []string{"backups"},
+			Verbs:     []string{"create", "get", "list", "update", "patch"},
+		},
+	}
+
+	// postgresql need more rules for patroni
+	if isPostgresqlCluster(transCtx) {
+		rules = append(rules, []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{""},
+				Resources: []string{"configmaps"},
+				Verbs:     []string{"create", "get", "list", "patch", "update", "watch", "delete"},
+			},
+			{
+				APIGroups: []string{""},
+				Resources: []string{"endpoints"},
+				Verbs:     []string{"create", "get", "list", "patch", "update", "watch", "delete"},
+			},
+			{
+				APIGroups: []string{""},
+				Resources: []string{"pods"},
+				Verbs:     []string{"get", "list", "patch", "update", "watch"},
+			},
+		}...)
+	}
+	role.Rules = append(role.Rules, rules...)
+}
+
+func isPostgresqlCluster(transCtx *ClusterTransformContext) bool {
```

Comment:
Better to implement an API to be compatible with postgresql

Reply:
The RBAC API integration is out of scope for this pull request but will be implemented in the next proposed update.